app/src/pages/UserPage/UserPage.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import { RestMethod, truncateMiddle } from '@kibalabs/core';
+import { useStringRouteParam } from '@kibalabs/core-react';
+import { Alignment, Box, Direction, Image, PaddingSize, Spacing, Stack, Text } from '@kibalabs/ui-react';
+
+import { CollectionToken } from '../../client/resources';
+import { TokenCard } from '../../components/TokenCard';
+import { useGlobals } from '../../globalsContext';
+import { Token, TokenCollection } from '../../model';
+import { usePageData } from '../../PageDataContext';
+import { loadTokenCollection } from '../../util';
+import { IHomePageData } from '../HomePage/getHomePageData';
+
+export const UserPage = (): React.ReactElement => {
+  const { notdClient, requester, projectId } = useGlobals();
+  const { data } = usePageData<IHomePageData>();
+  const accountAddress = useStringRouteParam('accountAddress');
+  const [holdings, setHoldings] = React.useState<Token[] | undefined | null>(undefined);
+  const [tokenCollection, setTokenCollection] = React.useState<TokenCollection | undefined>(data?.tokenCollection || undefined);
+  const loadMetadata = React.useCallback(async (): Promise<void> => {
+    const metadataResponse = await requester.makeRequest(RestMethod.GET, `${window.location.origin}/assets/${projectId}/metadatas.json`);
+    const loadedTokenCollection = loadTokenCollection(JSON.parse(metadataResponse.content) as Record<string, unknown>);
+    setTokenCollection(loadedTokenCollection);
+  }, [requester, projectId]);
+  React.useEffect((): void => {
+    loadMetadata();
+  }, [loadMetadata]);
+
+  const getCollectionHoldings = React.useCallback(async (shouldClear = false): Promise<void> => {
+    if (shouldClear) {
+      setHoldings(undefined);
+    }
+    if (!accountAddress) {
+      setHoldings(null);
+      return;
+    }
+    if (!tokenCollection) {
+      setHoldings(undefined);
+      return;
+    }
+    notdClient.getCollectionHoldings(tokenCollection.address, accountAddress).then((tokenTransfers: CollectionToken[]): void => {
+      const newOwnedTokens: Token[] = tokenTransfers.map((tokenTransfer: CollectionToken) => {
```

Comment:
you need to convert these to the corret type of token - i said this to you on slakc yesterday and you didnt respond!!